src/core/thread/mle_tlvs.hpp

```diff
@@ -109,7 +109,14 @@ class Tlv : public ot::Tlv
         kActiveDataset       = 24, ///< Active Operational Dataset TLV
         kPendingDataset      = 25, ///< Pending Operational Dataset TLV
         kDiscovery           = 26, ///< Thread Discovery TLV
-        kInvalid             = 255,
+
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+        kTimeRequest   = 252, ///< Time Request TLV
```

Comment:
Avoid wrapping these kinds of definitions to ensure that there are no symbol redefinitions even when `OPENTHREAD_CONFIG_ENABLE_TIME_SYNC` is 0.  I would also add to the comment that these are only used with `OPENTHREAD_CONFIG_ENABLE_TIME_SYNC`.

Reply:
Ok, Done.